src/openrct2/windows/loadsave.c

```diff
@@ -812,6 +821,10 @@ static void window_loadsave_select(rct_window *w, const char *path)
 			window_loadsave_invoke_callback(MODAL_RESULT_FAIL, path);
 		}
 	}
+	case (LOADSAVETYPE_LOAD | LOADSAVETYPE_IMAGE):
+		window_close_by_class(WC_LOADSAVE);
+		window_loadsave_invoke_callback(MODAL_RESULT_OK, path);
```

Comment:
`path` gets freed in `window_loadsave_close`, which is already gone through a call to `window_close_by_class`